src/index.ts

```diff
@@ -32,31 +32,25 @@ export type FieldGenerator =
   | OneOfGenerator
   | PerBuildGenerator;
 
-export type Field =
-  | string
-  | number
-  | null
-  | FieldGenerator
-  | { [x: string]: Field | any }
-  | any[];
+export type Field<T = any> = T | FieldGenerator | FieldsConfiguration<T>;
 
 export type FieldsConfiguration<FactoryResultType> = {
-  readonly [x in keyof FactoryResultType]: Field;
+  readonly [Key in keyof FactoryResultType]: Field<FactoryResultType[Key]>;
 };
 
-export interface Overrides {
-  [x: string]: Field;
-}
+export type Overrides<FactoryResultType = any> = {
+  [Key in keyof FactoryResultType]?: Field<FactoryResultType[Key]>;
+};
```

Comment:
This can likely be simplified to:
```suggestion
export type Overrides<FactoryResultType = any> = Partial<FieldsConfiguration<FactoryResultType>>
```